stable_baselines/a2c/a2c.py

```diff
@@ -335,6 +347,10 @@ def run(self):
                 maybe_ep_info = info.get('episode')
                 if maybe_ep_info is not None:
                     ep_infos.append(maybe_ep_info)
+                if info.get('valid_actions') is not None:
+                    action_mask = np.expand_dims(np.array(info.get('valid_actions'), dtype=np.bool),
+                                                            axis=0)
+                    mb_action_masks.append(action_mask)
```

Comment:
action_mask gets changed above when you run model.step.  Need to clone in reshape or here, otherwise, you end up with all but the last step corrupted.

Reply:
The action_mask passed into model.step needs to be in the format (n, mask), that is why I np.expand_dims. How does this corrupt the last step?